proxy/contracts/mainnet/Linker.sol

```diff
@@ -21,92 +21,76 @@
-        for (index = 0; index < length; index++) {
-            if (_depositBoxes[index] == depositBoxAddress) {
-                break;
-            }
-        }
-        if (index < length) {
-            if (index < length.sub(1)) {
-                _depositBoxes[index] = _depositBoxes[length.sub(1)];
-            }
-            _depositBoxes.pop();
-        }
+    function registerDepositBox(address newDepositBoxAddress) external onlyLinker {
+        _depositBoxes.add(newDepositBoxAddress);
+    }
+
+    function removeDepositBox(address depositBoxAddress) external onlyLinker {
+        _depositBoxes.remove(depositBoxAddress);
     }
 
-    function connectSchain(string calldata schainName, address[] calldata tokenManagerAddresses) external onlyOwner {
-        require(tokenManagerAddresses.length == _depositBoxes.length, "Incorrect number of addresses");
+    function connectSchain(string calldata schainName, address[] calldata tokenManagerAddresses) external onlyLinker {
+        require(tokenManagerAddresses.length == _depositBoxes.length(), "Incorrect number of addresses");
         for (uint i = 0; i < tokenManagerAddresses.length; i++) {
-            IDepositBox(_depositBoxes[i]).addTokenManager(schainName, tokenManagerAddresses[i]);
+            IDepositBox(_depositBoxes.at(i)).addTokenManager(schainName, tokenManagerAddresses[i]);
         }
         messageProxy.addConnectedChain(schainName);
     }
 
-    function unconnectSchain(string calldata schainName) external onlyOwner {
-        uint length = _depositBoxes.length;
+    function unconnectSchain(string calldata schainName) external onlyLinker {
+        uint length = _depositBoxes.length();
         for (uint i = 0; i < length; i++) {
-            IDepositBox(_depositBoxes[i]).removeTokenManager(schainName);
+            IDepositBox(_depositBoxes.at(i)).removeTokenManager(schainName);
         }
         messageProxy.removeConnectedChain(schainName);
     }
 
-    // function rechargeSchainWallet(bytes32 schainId, uint256 amount) external {
-    //     require(address(this).balance >= amount, "Not enough ETH to rechargeSchainWallet");
-    //     address walletsAddress = IContractManager(contractManagerOfSkaleManager).getContract("Wallets");
-    //     IWallets(payable(walletsAddress)).rechargeSchainWallet{value: amount}(schainId);
-    // }
-
     function hasDepositBox(address depositBoxAddress) external view returns (bool) {
-        uint index;
-        uint length = _depositBoxes.length;
-        for (index = 0; index < length; index++) {
-            if (_depositBoxes[index] == depositBoxAddress) {
-                return true;
-            }
-        }
-        return false;
+        return _depositBoxes.contains(depositBoxAddress);
     }
 
     function hasSchain(string calldata schainName) external view returns (bool connected) {
-        uint length = _depositBoxes.length;
+        uint length = _depositBoxes.length();
         connected = true;
```

Comment:
Looks like we could simplify it: 
`connected = messageProxy.isConnectedChain(schainName)`
and then remove 
`connected = connected && messageProxy.isConnectedChain(schainName);`